backend/PennCourses/docs_settings.py

```diff
@@ -866,53 +866,86 @@ def _get_responses(self, path, method):
         else:
             response_schema = item_schema
 
+        # Automatically add 403 response if authentication is required
+        auth_response_dict = (
+            {403: {"description": "Authentication credentials were not provided."}}
+            if IsAuthenticated in self.view.permission_classes
+            else {}
+        )
+
         if (
             path not in self.response_codes.keys()
             or method not in self.response_codes[path].keys()
             or not self.response_codes[path][method]
         ):
             if path in self.override_schema and method in self.override_schema[path].keys():
                 return {
+                    **{
+                        "200": {
+                            "description": "",
+                            **(
+                                {
+                                    "content": {
+                                        ct: {"schema": self.override_schema[path][method][200]}
+                                        for ct in self.response_media_types
+                                    }
+                                    if 200 in self.override_schema[path][method].keys()
+                                    else {
+                                        ct: {"schema": response_schema}
+                                        for ct in self.response_media_types
+                                    }
+                                }
+                            ),
+                        }
+                    },
+                    **auth_response_dict,
+                }
+            return {
+                **{
                     "200": {
+                        "content": {
+                            ct: {"schema": response_schema} for ct in self.response_media_types
+                        },
+                        # description is a mandatory property,
+                        # https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#responseObject  # noqa: E501
+                        # TODO: put something meaningful into it
                         "description": "",
+                    }
+                },
+                **auth_response_dict,
+            }
+        if path in self.override_schema and method in self.override_schema[path].keys():
+            return {
+                **{
+                    str(k): {
```

Comment:
same here. could you replace `k` and `v` with something a bit more specific? `k` seems to be the status code and `v` seems to be the message.

Reply:
Done.  Good point, will make sure to use more descriptive names in the future.